google/cloud/spanner/samples/pg_samples.cc

```diff
@@ -0,0 +1,371 @@
+  if (project_id.empty()) {
+    throw std::runtime_error("GOOGLE_CLOUD_PROJECT is not set or is empty");
+  }
+
+  auto random_instance = google::cloud::spanner_testing::PickRandomInstance(
+      generator, project_id,
+      "labels.samples:yes AND NOT name:/instances/test-instance-mr-");
+  if (!random_instance) {
+    throw std::runtime_error("Cannot find an instance to run the samples: " +
+                             random_instance.status().message());
+  }
+  auto const instance_id = *std::move(random_instance);
+  std::cout << "Running samples on " << instance_id << std::endl;
+
+  auto const database_id =
+      google::cloud::spanner_testing::RandomDatabaseName(generator);
+
+  google::cloud::spanner::Instance instance(project_id, instance_id);
+  google::cloud::spanner::Database database(instance, database_id);
+
+  google::cloud::spanner_admin::DatabaseAdminClient database_admin_client(
+      google::cloud::spanner_admin::MakeDatabaseAdminConnection());
+
+  SampleBanner("spanner_postgresql_create_database");
+  samples::CreateDatabase(database_admin_client, database);
+
+  SampleBanner("spanner_postgresql_create_tables");
+  samples::CreateTables(database_admin_client, database);
+
+  SampleBanner("spanner_postgresql_add_column");
+  samples::AddColumn(database_admin_client, database);
+
+  SampleBanner("spanner_postgresql_create_storing_index");
+  samples::CreateStoringIndex(database_admin_client, database);
+
+  google::cloud::spanner::Client client(
+      google::cloud::spanner::MakeConnection(database));
+
+  SampleBanner("spanner_insert_data");
+  samples::InsertData(client);
+
+  SampleBanner("spanner_postgresql_query_with_parameter");
+  samples::QueryWithParameter(client);
+
+  SampleBanner("spanner_postgresql_dml_getting_started_update");
+  samples::DmlGettingStartedUpdate(client);
+
+  /*
+   * Remaining samples to add:
+   *   spanner_postgresql_batch_dml
+   *   spanner_postgresql_case_sensitivity
+   *   spanner_postgresql_cast_data_type
+   *   spanner_postgresql_create_clients
+   *   spanner_postgresql_dml_with_parameters
+   *   spanner_postgresql_functions
+   *   spanner_postgresql_information_schema
+   *   spanner_postgresql_interleaved_table
+   *   spanner_postgresql_order_nulls
+   *   spanner_postgresql_partitioned_dml
+   *   spanner_postgresql_numeric_data_type
+   */
+
+  SampleBanner("spanner_drop_database");
+  samples::DropDatabase(database_admin_client, database);
+
+  return 0;
+}
+
+}  // namespace
+
+int main(int ac, char* av[]) try {
+  const auto* const emulator_host = "SPANNER_EMULATOR_HOST";
+  if (google::cloud::internal::GetEnv(emulator_host).has_value()) {
+    return 0;  // emulator does not support PostgreSQL dialect
+  }
+  const auto* const auto_run = "GOOGLE_CLOUD_CPP_AUTO_RUN_EXAMPLES";
+  if (google::cloud::internal::GetEnv(auto_run).value_or("") == "yes") {
+    return RunAll();
+  }
+  std::string program(ac ? (ac--, *av++) : "pg_samples");
```

Comment:
TIL `argc` can be 0

Reply:
Yeah.  It can be instructive to try executing common programs with no arguments and seeing how many survive the ordeal.